pkgs/development/compilers/cudatoolkit/default.nix

```diff
@@ -56,5 +57,19 @@ in rec {
     gcc = gcc9;
   };
 
+  cudatoolkit_11_3 = common {
+    version = "11.3.1";
+    url = "https://developer.download.nvidia.com/compute/cuda/11.3.1/local_installers/cuda_11.3.1_465.19.01_linux.run";
+    sha256 = "sha256:0d19pwcqin76scbw1s5kgj8n0z1p4v1hyfldqmamilyfxycfm4xd";
+    gcc = gcc9;
+  };
+
+  cudatoolkit_11_4 = common {
+    version = "11.4.1";
+    url = "https://developer.download.nvidia.com/compute/cuda/11.4.1/local_installers/cuda_11.4.1_470.57.02_linux.run";
+    sha256 = "sha256:0180pb1zfajb9l6blr467xkx01yp3snfwm2xix8x52crf6d36v6x";
+    gcc = gcc11;
+  };
+
   cudatoolkit_11 = cudatoolkit_11_2;
```

Comment:
```suggestion
  cudatoolkit_11 = cudatoolkit_11_4;
```

Reply:
I think we should add a [compatible CuDNN](https://docs.nvidia.com/deeplearning/cudnn/support-matrix/index.html) first before updating `cudatoolkit_11`, since otherwise `cudatoolkit_11` and  `cudnn_cudatoolkit_11` are not compatible.

Reply:
I'd be happy to do that as well. Either in this pr or in a follow-up one.